feature-libs/user/profile/components/register/register-component.service.ts

```diff
@@ -46,4 +49,27 @@ export class RegisterComponentService {
       GlobalMessageType.MSG_TYPE_CONFIRMATION
     );
   }
+
+  /**
+   * Get if any additional consents needs to be provided during registration
+   * In core feature, no additional consents are necessary during registration.
+   * In integration scenarios, eg: cdc, this method will be overridden to return
+   * necessary cdc consents
+   */
+  getAdditionalConsents(): {
+    template: ConsentTemplate;
+    required: boolean;
+  }[] {
+    return [];
+  }
+
+  /**
+   * Generate form control if any additional consents that needs to be provided during registration
+   * In core feature, no additional consents are necessary during registration.
+   * In integration scenarios, eg: cdc, this method will be overridden to return
+   * form controls for necessary cdc consents
+   */
+  generateAdditionalConsentsFormControl(): UntypedFormArray {
+    return this.fb.array([]);
+  }
```

Comment:
We need to gracefully fallback, when `this.fb` is undefined.


```diff
-  generateAdditionalConsentsFormControl(): UntypedFormArray {
-    return this.fb.array([]);
  generateAdditionalConsentsFormControl(): UntypedFormArray | undefined {
    return this.fb?.array([]) ?? undefined;
  }
  }
```

Reply:
done